internal/kms/kes.go

```diff
@@ -104,7 +104,7 @@ func NewWithConfig(config Config) (KMS, error) {
 				}
 				sameCert := true
 				for i, b := range certificate.Certificate {
-					if !bytes.Equal(b, prevCertificate.Certificate[i]) {
+					if len(prevCertificate.Certificate) > 0 && !bytes.Equal(b, prevCertificate.Certificate[i]) {
```

Comment:
1) It can still crash when adding more. 2) If `len(prevCertificate.Certificate) == 0` it will never reload.

Entire block suggestion:

```Go
	sameCert := len(certificate.Certificate) == len(prevCertificate.Certificate)
	for i, b := range certificate.Certificate {
		if !sameCert {
			break
		}
		sameCert = bytes.Equal(b, prevCertificate.Certificate[i])
	}
```